src/prefs.js

```diff
@@ -235,39 +235,36 @@ export default class SensorsPreferences
         return row;
     }
 
-    _getSensorsLabels() {
-        let sensors_cmd = Utilities.detectSensors();
+    async _getSensorsLabels() {
+        const sensors_cmd = Utilities.detectSensors();
         if(sensors_cmd) {
-            let [result, sensors_output] = GLib.spawn_command_line_sync(sensors_cmd.join(' '));
-            if(result)
+            const [sensors_output, hasError] = await SubProc.runCommandAsync(sensors_cmd);
+            if(sensors_output)
             {
-                const decoder = new TextDecoder();
-                let output = decoder.decode(sensors_output);
-                let tempInfo = Utilities.parseSensorsOutput(output,Utilities.parseSensorsTemperatureLine);
+                const tempInfo = Utilities.parseSensorsOutput(sensors_output,Utilities.parseSensorsTemperatureLine);
                 this._appendMultipleItems(tempInfo);
 
                 if (this._display_fan_rpm){
-                    let fanInfo = Utilities.parseSensorsOutput(output,Utilities.parseFanRPMLine);
+                    const fanInfo = Utilities.parseSensorsOutput(sensors_output,Utilities.parseFanRPMLine);
                     this._appendMultipleItems(fanInfo);
                 }
                 if (this._display_voltage){
-                    let voltageInfo = Utilities.parseSensorsOutput(output,Utilities.parseVoltageLine);
+                    const voltageInfo = Utilities.parseSensorsOutput(sensors_output,Utilities.parseVoltageLine);
                     this._appendMultipleItems(voltageInfo);
                 }
             }
         }
     }
 
-    _getHddTempLabels() {
-        let hddtemp_cmd = Utilities.detectHDDTemp();
+    async _getHddTempLabels() {
+        const hddtemp_cmd = Utilities.detectHDDTemp();
         if(hddtemp_cmd){
-            let [result, hddtemp_output] = GLib.spawn_command_line_sync(hddtemp_cmd.join(' '))
-            if(result){
-                const decoder = new TextDecoder();
-                let hddTempInfo = Utilities.parseHddTempOutput(
-                    decoder.decode(hddtemp_output),
```

Comment:
The communicate_utf8_async method returns suitable text so TextDecoder no longer required